tensorflow/core/kernels/mlir_generated/build_defs.bzl

```diff
@@ -101,29 +137,41 @@ def _gen_kernel_image_hdr_impl(ctx):
         mnemonic = "bin2c",
     )
 
+def _gen_kernel_image_hdr_impl_rocm(ctx):
+    hsacos = ctx.attr.input[CubinInfo].hsacos
+    ctx.actions.run_shell(
+        outputs = [ctx.outputs.out],
+        inputs = hsacos,
+        command = "echo 'const std::string %s = \"/root/tensorflow/%s\";' > %s" % (ctx.attr.symbol, hsacos[0].path, ctx.outputs.out.path),
```

Comment:
I guess this is the prototype aspect of this PR? :)
In the end we probably need to find a different way, because the assumption that the code root is always /root/tensorflow is probably not valid?

Reply:
yes.  The prototype aspect primarily involves a couple of things

1. We need to implement the equivalent of `cuda_gpu_architectures` for ROCm. That will involve passing that information in via rocm_configure.bzl / local_config_rocm. I expect this to be present in the next iteration as well

2. We need to figure out how to implement the equivalent of `fatbinary` and `bin2c` on the ROCm platform. The prototype code
 * supports generating HSACOs (ROCm GPU Binaries) for a single architecture only (gfx900) 
  *need something like `fatbinary` to support more than arch 
 * needs to pass the harcoded file name of the HSACO file, because of the lack of a `bin2c` type utility

We are still looking into how to make this part work on ROCm. Any suggestions / feedback is welcome.